drivers/sensor/bmi323/bmi323.h

```diff
@@ -0,0 +1,196 @@
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_LVL_SIZE		(0x01)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_LVL_VAL_ACT_LOW	(0x00)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_LVL_VAL_ACT_HIGH	(0x01)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_OD_OFFSET		(0x01)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_OD_SIZE		(0x01)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_OD_VAL_PUSH_PULL	(0x00)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_OD_VAL_OPEN_DRAIN (0x01)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_OUTPUT_EN_OFFSET	(0x02)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_OUTPUT_EN_SIZE	(0x01)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_OUTPUT_EN_VAL_DIS (0x00)
+#define IMU_BOSCH_BMI323_REG_IO_INT_CTRL_INT1_OUTPUT_EN_VAL_EN	(0x01)
+
+#define IMU_BOSCH_BMI323_REG_INT_CONF (0x39)
+
+#define IMU_BOSCH_BMI323_REG_INT_CONF_INT_LATCH_OFFSET		(0x01)
+#define IMU_BOSCH_BMI323_REG_INT_CONF_INT_LATCH_SIZE		(0x01)
+#define IMU_BOSCH_BMI323_REG_INT_CONF_INT_LATCH_VAL_NON_LATCHED (0x00)
+#define IMU_BOSCH_BMI323_REG_INT_CONF_INT_LATCH_VAL_LATCHED	(0x01)
+
+#define IMU_BOSCH_BMI323_REG_INT_MAP1 (0x3A)
+
+#define IMU_BOSCH_BMI323_REG_INT_MAP1_MOTION_OUT_OFFSET	  (0x02)
+#define IMU_BOSCH_BMI323_REG_INT_MAP1_MOTION_OUT_SIZE	  (0x02)
+#define IMU_BOSCH_BMI323_REG_INT_MAP1_MOTION_OUT_VAL_DIS  (0x00)
+#define IMU_BOSCH_BMI323_REG_INT_MAP1_MOTION_OUT_VAL_INT1 (0x01)
+#define IMU_BOSCH_BMI323_REG_INT_MAP1_MOTION_OUT_VAL_INT2 (0x02)
+
+#define IMU_BOSCH_BMI323_REG_INT_MAP2 (0x3B)
+
+#define IMU_BOSCH_BMI323_REG_INT_MAP2_ACC_DRDY_INT_OFFSET   (0x0A)
+#define IMU_BOSCH_BMI323_REG_INT_MAP2_ACC_DRDY_INT_SIZE	    (0x02)
+#define IMU_BOSCH_BMI323_REG_INT_MAP2_ACC_DRDY_INT_VAL_DIS  (0x00)
+#define IMU_BOSCH_BMI323_REG_INT_MAP2_ACC_DRDY_INT_VAL_INT1 (0x01)
+#define IMU_BOSCH_BMI323_REG_INT_MAP2_ACC_DRDY_INT_VAL_INT2 (0x02)
+
+#define IMU_BOSCH_BMI323_REG_FEATURE_CTRL (0x40)
+
+#define IMU_BOSCH_BMI323_REG_FEATURE_CTRL_ENABLE_OFFSET	 (0x00)
+#define IMU_BOSCH_BMI323_REG_FEATURE_CTRL_ENABLE_SIZE	 (0x01)
+#define IMU_BOSCH_BMI323_REG_FEATURE_CTRL_ENABLE_VAL_DIS (0x00)
+#define IMU_BOSCH_BMI323_REG_FEATURE_CTRL_ENABLE_VAL_EN	 (0x01)
+
+#define IMU_BOSCH_BMI323_REG_CMD		    (0x7E)
+#define IMU_BOSCH_BMI323_REG_CMD_CMD_OFFSET	    (0x00)
+#define IMU_BOSCH_BMI323_REG_CMD_CMD_SIZE	    (0x10)
+#define IMU_BOSCH_BMI323_REG_CMD_CMD_VAL_SOFT_RESET (0xDEAF)
+
+#define IMU_BOSCH_BMI323_REG_MASK(reg, field)                                                      \
+	(BIT_MASK(IMU_BOSCH_BMI323_REG_##reg##_##field##_SIZE)                                     \
+	 << IMU_BOSCH_BMI323_REG_##reg##_##field##_OFFSET)
+
+#define IMU_BOSCH_BMI323_REG_VALUE(reg, field, val)                                                \
+	(IMU_BOSCH_BMI323_REG_##reg##_##field##_VAL_##val                                          \
+	 << IMU_BOSCH_BMI323_REG_##reg##_##field##_OFFSET)
+
+#define IMU_BOSCH_BMI323_REG_VALUE_GET_FIELD(reg_value, reg, field)                                \
```

Comment:
I will update it to use the existing macros :)

Reply:
After looking into updating the defines, the Macros in Utils take a quite different approach to the one I have used for defining and using the registers, which requires changing most of them in both header and source files. Since its not a blocker, I will leave the defines as they are in this case, and use the utils macros moving forward :)